### PR TITLE
Make UI less dependent on the rest of uTox.

### PR DESCRIPTION
**Pull request by robinlinden**
Moves a bunch of scale and prettify text functions from main.h to ui.h
Moves draw function declarations from main.h to ui/draw.h
Adds a few forward declares
Removes a deprecated drawing function
Some minor cleanup of files I touched (mostly adding const, C99ing loops, and lowering the scope of variables.)

---
This change is [Reviewable](https://reviewable.io/reviews/utox/utox/610)

### CODE REVIEW

File: src/ui.h [hunk 1]
```diff
@@ -48,7 +48,7 @@ struct panel {
     PANEL **child;
 };
 
-typedef struct {
+typedef struct maybe_i18nal_string {
```
Comment (robinlinden) [inline src/ui.h]: Made this change so I could forward declare this struct.

**Comment by endoffile78** [comment]
:lgtm:

---

Reviewed 26 of 26 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610#-:-KaKYAwy0NlwTfAAJYYq:bnfp4nl)*

**Comment by cebe** [comment]
There are now draw.h and drawing.c maybe choose a consistent name?

---

Reviewed 25 of 26 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/theme.c, line 516 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLrinNhAFD2MjV8kMg:-KaM_11ghfNiX3W-_baE:bndsa4f) ([raw file](https://github.com/utox/utox/blob/c77da56e7e54ea9d8ef39af295dc2e6198fa91b1/src/theme.c#L516)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

we do need a way to debug this. I'm not saying bring it back, but a comment is probably important.
</blockquote></details>

then just re-add it when you need it?

---

*[src/ui/CMakeLists.txt, line 10 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaM_yf_aeE6fv4fXfbn:-KaM_yfaUlYfDbOY0UaM:bhlzuy0) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/CMakeLists.txt#L10)):*
> ```CMake
>     contextmenu.c
>     contextmenu.h
>     draw.h
> ```

there is no need to put headers in the cmake file, they do not need their own .o object file. Cmake detects dependencies of .c files on headers itself.

---

*[src/ui/svg.c, line 710 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLqNg18mz_hGuETZQq:-KaMbW3DEYOHWAGSP9Xb:bqnm9i6) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/svg.c#L710)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

this can be an assert, it's literally only helpful to developers
</blockquote></details>

:+1: we should have a separate header to include logging anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610#-:-KaMcldIoJStoTpaAvre:b6nejit)*

**Comment by GrayHatter** [comment]
re drawing or draw

make it draw.*
I'll rename the other files in the windows_popup branch tomorrow.

**Comment by robinlinden** [comment]
I'll update the header guards on this one to be UI_DRAW_H.

---

Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/theme.c, line 516 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLrinNhAFD2MjV8kMg:-KaNP4qgNdSN8oOUTuHl:b-6bd5gw) ([raw file](https://github.com/utox/utox/blob/c77da56e7e54ea9d8ef39af295dc2e6198fa91b1/src/theme.c#L516)):*
<details><summary><i>Previously, cebe (Carsten Brandt) wrote…</i></summary><blockquote>

then just re-add it when you need it?
</blockquote></details>

We should probably look over where we have our logging placed. Including main.h just for logging isn't great. If you want, I'll add it back for now since this file is still including both main.h and native_main.h. 

Theme.c is going to be moved to be part of the UI ~soon since it has nothing to do with the actual logic of uTox and we can look over this then.

---

*[src/ui/buttons.c, line 12 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLpkVcOAmyCYPxpw7h:-KaNQp34NtdbaGMnMXvR:b758k4g) ([raw file](https://github.com/utox/utox/blob/c77da56e7e54ea9d8ef39af295dc2e6198fa91b1/src/ui/buttons.c#L12)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

BTW good catch!
</blockquote></details>

Thank you. :D

---

*[src/ui/CMakeLists.txt, line 10 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaM_yf_aeE6fv4fXfbn:-KaNQv6kpijsj6wCHKFh:b-kwg7t7) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/CMakeLists.txt#L10)):*
<details><summary><i>Previously, cebe (Carsten Brandt) wrote…</i></summary><blockquote>

there is no need to put headers in the cmake file, they do not need their own .o object file. Cmake detects dependencies of .c files on headers itself.
</blockquote></details>

Cool. I didn't know this. Deleted in the next commit.

---

*[src/ui/draw_helpers.c, line 261 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLq-3GJCnM5KvKYrml:-KaNR3qbvlhyzgy8moeW:bidordm) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/draw_helpers.c#L261)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

at your option, drop the default, and typedef the enum
</blockquote></details>

That's a good idea. Will do.

---

*[src/ui/svg.c, line 710 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLqNg18mz_hGuETZQq:-KaNR6rGKpjgAB8Iy28r:b-w7jvp3) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/svg.c#L710)):*
<details><summary><i>Previously, cebe (Carsten Brandt) wrote…</i></summary><blockquote>

:+1: we should have a separate header to include logging anyway.
</blockquote></details>

CeBe++. Will add the assert for now.

---

*[src/xlib/event.c, line 7 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLr07JYS2L8yoaKWvQ:-KaNRJkfdpTs4hcnw0bg:b67xdci) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/xlib/event.c#L7)):*
<details><summary><i>Previously, GrayHatter (Gregory Mullen) wrote…</i></summary><blockquote>

it should be changed by changing the scope of enddraw(), that's more of a native thing than a drawing thing.
</blockquote></details>

It's still nothing that the main logic of uTox should have to use. Drawing is strictly part of the UI. All native_draw stuff will be moved to the UI and then uTox's core could be used for e.g. a cmdline Tox client.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610#-:-KaNVPyc-PaI1ybqZJFv:b-5m0rin)*

**Comment by cebe** [comment]
Reviewed 1 of 26 files at r1.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610)*

**Comment by GrayHatter** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/theme.c, line 516 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLrinNhAFD2MjV8kMg:-KaOnrMA2hHaR-BerS8B:b8mbmca) ([raw file](https://github.com/utox/utox/blob/c77da56e7e54ea9d8ef39af295dc2e6198fa91b1/src/theme.c#L516)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

We should probably look over where we have our logging placed. Including main.h just for logging isn't great. If you want, I'll add it back for now since this file is still including both main.h and native_main.h. 

Theme.c is going to be moved to be part of the UI ~soon since it has nothing to do with the actual logic of uTox and we can look over this then.
</blockquote></details>

agreed, this is non-blocking

---

*[src/xlib/event.c, line 7 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLr07JYS2L8yoaKWvQ:-KaOpgdskYbOafDMdOds:b-hrk381) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/xlib/event.c#L7)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

It's still nothing that the main logic of uTox should have to use. Drawing is strictly part of the UI. All native_draw stuff will be moved to the UI and then uTox's core could be used for e.g. a cmdline Tox client.
</blockquote></details>

I agree, but it's not a drawing thing either. It's coping what the native functions created to the video driver.

Either way, I'm pretty sure the way you want to do it will be correct. I just meant to say, that function is more of a window_ function than a draw_ function.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610#-:-KaOnmy1MIY3_oSoPgHE:bnfp4nl)*

**Comment by cebe** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610#-:-KaOweiKorEaocbGHD8s:bnfp4nl)*

**Comment by cebe** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/ui/svg.c, line 710 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLqNg18mz_hGuETZQq:-KaP6l-gx_AbPhusY1B_:b-wn651c) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/svg.c#L710)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

CeBe++. Will add the assert for now.
</blockquote></details>

I've prepared a PR for separating out logging.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610)*

**Comment by GrayHatter** [comment]
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---

*[src/ui/svg.c, line 710 at r2](https://reviewable.io:443/reviews/utox/utox/610#-KaLqNg18mz_hGuETZQq:-KaP8nMkrt0VcKvpIwcb:b-6waudk) ([raw file](https://github.com/utox/utox/blob/82952c16859a0ad78c7487258f679121726ecf85/src/ui/svg.c#L710)):*
<details><summary><i>Previously, cebe (Carsten Brandt) wrote…</i></summary><blockquote>

I've prepared a PR for separating out logging.
</blockquote></details>

already approved too

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/utox/utox/610)*

**Pull request merged**